Fix scroll when loads the client bundle

gabrielrtakeda opened the pull request:
# Description
When the URL have a hash pointing to an element id,
scroll to it on the client bundle. The problem was
that the initial render responded by server did the
scroll but, when the client bundle was loaded, the
scroll was back to the top.

# Related issues
- Wierd scroll to top after first load from mobilization #478

# How to test

**Fragment case**
- Access a mobilization with a hash on URL that points to a block e.g. http://www.39-layout-minha-sampa-versao-02.staging.bonde.org/#block-264
- After initial render, the scroll should not back to the top.

**Initial render scroll case**
- Access a mobilization public view and scroll until page are loading
- After the page assets finish loading, the page scroll must still the same as scrolled before.

gabrielrtakeda committed:
Fix scroll when loads the client bundle fixes #478

When the URL have a hash pointing to an element id,
scroll to it on the client bundle. The problem was
that the initial render responded by server did the
scroll but, when the client bundle was loaded, the
scroll was back to the top.

lpirola commented:
@gabrielrtakeda this fix only apply to users when they came to a page with a fragment, right? I was wondering if the erro could be something with client re-render after load server side page. What do you think about this?

gabrielrtakeda commented:
@lpirola I think we can't prevent to re-render the page when the client bundle loads. Maybe one thing we can do is to avoid the scroll on initial render provided by the SSR.

gabrielrtakeda commented:
We can't "pass" the current scroll position on the **initial render** to the **client bundle** because both are distinct bundles.

gabrielrtakeda commented:
Or may we can. I will try an idea that came up to me now 🙂

gabrielrtakeda committed:
Pass the initial render scroll position fixes #478

Prevent to browser rescroll to top when the client bundle
finish to load, passing the scroll position of the initial
render to a global variable that the client bundle can get
when it mounts.

gabrielrtakeda commented:
@lpirola I think I get this to work.

gabrielrtakeda commented:
@lpirola After staging deploy finish, can you test it and validate if the solution fixes the bug?

lpirola commented:
@gabrielrtakeda works like a charm! Go live!

gabrielrtakeda commented:
Yay! Go go go! :tada:
